File: src/services/fonts.ts
import { invoke } from "@tauri-apps/api";

interface Font {
    font: string;
    small_text: number;
    regular_text: number;
    big_text: number;
    regular_title: number;
    big_title: number;
}

export async function get_fonts(): Promise<Font> {
    let fonts: Font = await invoke("get_fonts_file");
    return fonts;
}

File: src-tauri/src/config/default/fonts.rs
use crate::config::Fonts;

pub fn default_fonts() -> Fonts {
    let default_fonts = Fonts {
        font: "JetBrains Mono".to_string(),
        small_text: ".75rem".to_string(),
        regular_text: "1rem".to_string(),
        big_text: "1.25rem".to_string(),
        regular_title: "2rem".to_string(),
        big_title: "2.5rem".to_string(),
    };

    default_fonts
}
